cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/adapter/AdapterAlreadyExistsException.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.internal.app.runtime.adapter;
+
+/**
+ * Thrown when Adapter creation is requested, but an adapter already exists.
+ */
+public class AdapterAlreadyExistsException extends Exception {
```

Comment:
We create an Exception here and also another one in client. We try to consolidate it. We should move everything to common module. We can then throw generic ```AlreadyExistsException```